src/Cookbook.js

```diff
@@ -4,19 +4,37 @@ class Cookbook {
     this.ingredients = ingredientsData;
   }
 
-  calculateCost(recipes) {
-
+  calculateCost(recipe) { 
+    let costInDollars = recipe.ingredients.reduce((acc, specificIngredient) => {
+      this.ingredients.forEach((ingredient) => {
+        if (specificIngredient.id === ingredient.id) {
+          acc += ingredient.estimatedCostInCents
+        }
+      })
+      return acc;
+    }, 0)
+    costInDollars /= 100;
+    return costInDollars.toLocaleString(undefined, { 
+      style: "currency", 
+      currency: "USD"
+    })
   }
+ 
 
   findRecipe(searchTerm) {
-    
-  }
-  
+    return this.recipes.filter(recipe => {
+      return recipe.ingredients.find(ingredient => {
+        let ingredientData = this.ingredients.find(element => element.id === ingredient.id);
+
+        return (ingredientData.name.includes(searchTerm)) ||
+        (recipe.name.includes(searchTerm))
+      });
+    })
+  } 
+
```

Comment:
Love how you did this method by using the ingredientData variable for the second find loop!